job/replay_validator.go

```diff
@@ -0,0 +1,133 @@
+package job
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/odpf/optimus/core/tree"
+	"github.com/odpf/optimus/models"
+	"github.com/odpf/optimus/store"
+)
+
+var (
+	// ReplayStatusToValidate signifies list of status to be used when checking active replays
+	ReplayStatusToValidate = []string{models.ReplayStatusInProgress, models.ReplayStatusAccepted, models.ReplayStatusReplayed}
+)
+
+type ReplayValidator interface {
+	Validate(context.Context, store.ReplaySpecRepository, models.ReplayRequest, *tree.TreeNode) error
+}
```

Comment:
We should move this to `replay_manager.go`

Reply:
moved